app/Http/Controllers/Admin/CategoryController.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Models\Category;
+use Illuminate\Http\Request;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\CategoryRequest;
+
+class CategoryController extends Controller
+{
+    public function index()
+    {
+        $categories = Category::all();
+
+        return view('admin.category.index', compact('categories'));
+    }
+
+    public function create()
+    {
+        $categoriesParent = Category::where('parent_id', '=', 0)->get();
+
+        return view('admin.category.create', compact('categoriesParent'));
+    }
+
+    public function store(CategoryRequest $request)
+    {
+        $categories = new Category;
+        $categories->name = $request->name;
+        $categories->parent_id = $request->parent_id;
+        $categories->save();
+
+        return redirect()->route('category.index')->with('success', trans('category.addSuccess'));
+    }
+
+    public function edit($category)
+    {
+        $category = Category::find($category);
+        $categoriesParent = Category::where('parent_id', '=', 0)->get();
+
+        return view('admin.category.update', compact('category', 'categoriesParent'));
```

Comment:
dòng 38 đã lấy ra toàn bộ cate. dòng 39 em lấy cate từ kết quả bên trên để tối ưu nhé

Reply:
dạ dòng 38 e chỉ lấy cate theo id,, còn dòng 39 e lấy những parent cate thôi ạ